### PR TITLE
Handle additional TX status (Failure), add network-name parameter

### PR DESCRIPTION
**Pull request by andreibancioiu**
- Add a new status: `Failure`
 - For invalid transactions emit two additional operations: the failed transfers
 - Bit of redesign around `chain ID`. We're using rosetta's terminology for CLI arguments: `network-name` & `network-id`. 
 - Update reference `rosetta-sdk-go=v0.8.1`
 - ~~Update reference `elrond-go=v1.3.42`~~ (postponed to future PR)

Additionally, apply a workaround to overcome the rapid exhaustion of TCP sockets under heavy load. References:
 - https://github.com/golang/go/issues/16012
 - https://superuser.com/questions/173535/what-are-close-wait-and-time-wait-states
 - https://vincent.bernat.ch/en/blog/2014-tcp-time-wait-state-linux
 - https://learn.microsoft.com/en-us/answers/questions/230227/time-wait-from-netstat.html

### CODE REVIEW

File: server/provider/networkProvider_test.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ func TestNewNetworkProvider(t *testing.T) {
 		ObservedProjectedShardIsSet: true,
 		ObserverUrl:                 "http://my-observer:8080",
 		ObserverPubkey:              "abba",
-		ChainID:                     "T",
+		ChainID:                     "devnet",
```
Comment (iulianpascalau) [inline server/provider/networkProvider_test.go:23]: The old value was more relevant, testing wise

**Review by AdoAdoAdo — DISMISSED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

File: server/provider/networkProvider_test.go (hunk 1 above)
Comment (andreibancioiu) [reply server/provider/networkProvider_test.go:23]: Actually I've made a mistake - we need to have two separate concepts: chain ID and chain name. I will repair this tomorrow.

File: cmd/rosetta/cli.go [hunk 2]
```diff
@@ -74,10 +74,16 @@ VERSION:
...
-		Usage: "Specifies the Chain ID.",
-		Value: "1",
+	cliFlagNetworkID = cli.StringFlag{
+		Name:     "network-id",
+		Usage:    "Specifies the network ID (e.g. 1, D, T)",
+		Required: true,
+	}
+
+	cliFlagNetworkName = cli.StringFlag{
+		Name:     "network-name",
```
Comment (andreibancioiu) [inline cmd/rosetta/cli.go:84]: We use rosetta's terminology here: blockchain name = elrond, network name = mainnet / devnet / testnet.

File: cmd/rosetta/cli.go [hunk 3]
```diff
@@ -74,10 +74,16 @@ VERSION:
 		Value: strings.Repeat("0", 64),
 	}
 
-	cliFlagChainID = cli.StringFlag{
-		Name:  "chain-id",
-		Usage: "Specifies the Chain ID.",
-		Value: "1",
+	cliFlagNetworkID = cli.StringFlag{
+		Name:     "network-id",
```
Comment (andreibancioiu) [inline cmd/rosetta/cli.go:78]: Necessary for transaction construction. This is, actually, the chain ID.

File: cmd/rosetta/main.go [hunk 4]
```diff
@@ -15,6 +15,14 @@ import (
 )
 
 func main() {
+	// This is a workaround, so that TCP sockets opened by rosetta (as a TCP client) to read data from the observer can be reused more easily,
```
Comment (andreibancioiu) [inline cmd/rosetta/main.go:18]: Perhaps we should also investigate the closing of HTTP requests on Observer's side, just to ensure it's all right.

Comment (andreibancioiu) [reply cmd/rosetta/main.go:18]: Please review a bit the explanation - is it correct?

Also see:
 - https://github.com/golang/go/issues/16012
 - https://vincent.bernat.ch/en/blog/2014-tcp-time-wait-state-linux

File: version/constants.go [hunk 5]
```diff
@@ -5,9 +5,9 @@ const (
 	RosettaVersion = "v1.4.12"
 
 	// RosettaMiddlewareVersion is the version of this package (application)
-	RosettaMiddlewareVersion = "v0.2.5"
+	RosettaMiddlewareVersion = "v0.2.6"
 
 	// NodeVersion is the canonical version of the node runtime
 	// TODO: We should fetch this from node/status.
-	NodeVersion = "rc/2022-july"
+	NodeVersion = "v1.3.42"
```
Comment (iulianpascalau) [inline version/constants.go:12]: if you set this to v1.3.42 please update go.mod to fetch the elrond-go v1.3.42 for consistency reasons

**Review by AdoAdoAdo — APPROVED** [review]

File: version/constants.go (hunk 5 above)
Comment (andreibancioiu) [reply version/constants.go:12]: `NodeVersion` is the version of the Observer to which rosetta connects, not the referenced version of elrond-go (used as a library).

However, the update in go.mod will be done as soon as possible, in a separate PR.

:pray:

**Pull request merged**